fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -4440,6 +4441,16 @@ void DDTeamCollection::evaluateTeamQuality() const {
 			varTeams += (stc - teamsPerServer) * (stc - teamsPerServer);
 			// Use zoneId as server's machine id
 			machineTeams[info->getLastKnownInterface().locality.zoneId()] += stc;
+			// Check invariant: if latest buildTeam succeeds, then each server must have at least
+			// targetTeamNumPerServer serverTeams
+			// lastBuildTeamsFailed is set only when (1) machine count is less than configured team size;
+			// (2) Not find any server team candidates when creating server team; (3) failed to add machine team
+			if (SERVER_KNOBS->DD_VALIDATE_SERVER_TEAM_COUNT_AFTER_BUILD_TEAM && !lastBuildTeamsFailed &&
```

Comment:
Can there be cases that build team always fails for some reason? If so, do we have signals of that has happened?

Reply:
Great idea!

Reply:
There are three cases that build team can be failed: 
1. BuildTeamsNotEnoughUniqueMachines trace event
2. DataDistributionBuildTeams trace event with Reason "Unable to make desired machine Teams"
3. BuildTeamsNotFindAnyServerTeam (new)